Sources/ReactiveSwift/Atomic.swift

```diff
@@ -123,43 +127,135 @@ internal struct UnsafeAtomicState<State: RawRepresentable>: AtomicStateProtocol
 #endif
 }
 
-final class PosixThreadMutex: NSLocking {
-	private var mutex = pthread_mutex_t()
+/// A reference counted version of `UnsafeUnfairLock`.
+internal final class UnfairLock {
+	let _lock: UnsafeUnfairLock
 
-	init() {
-		let result = pthread_mutex_init(&mutex, nil)
-		precondition(result == 0, "Failed to initialize mutex with error \(result).")
+	internal init(label: String = "") {
+		_lock = UnsafeUnfairLock(label: label)
 	}
 
 	deinit {
-		let result = pthread_mutex_destroy(&mutex)
-		precondition(result == 0, "Failed to destroy mutex with error \(result).")
+		_lock.destroy()
+	}
+
+	internal func lock() {
+		_lock.lock()
+	}
+
+	internal func unlock() {
+		_lock.unlock()
+	}
+
+	internal func `try`() -> Bool {
+		return _lock.try()
+	}
+}
+
+@_fixed_layout
```

Comment:
Is this a required part of this change? We should stay away from unsupported attributes if at all possible.